File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        # kth from left to right
        # do an inorder traversal 
        arr = self.getInorder(root) 
        return arr[k - 1]

    def getInorder(self, node): 
        if not node: 
            return []
        else: 
            return self.getInorder(node.left) + [node.val] + self.getInorder(node.right)

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        return getDepth(root, 0);
    }

    public int getDepth(TreeNode n, int curLength) {
        if (n != null) {
            int left = getDepth(n.left, curLength + 1);
            int right = getDepth(n.right, curLength + 1);
            return left > right ? left : right;
        } else {
            return curLength;
        }
    }
}
